.github/workflows/validation.yaml

```diff
@@ -94,3 +99,31 @@ jobs:
       - name: Run lint script
         run: npm run lint
         shell: bash
+
+  run-tests:
+    name: Run optional tests
+    runs-on: ubuntu-latest
+    if: inputs.npmTestScripts != ''
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+          token: ${{ secrets.GIT_TOKEN_BASIC || github.token }}
+
+      - uses: actions/setup-node@v3
+        with:
+          node-version: lts/*
+
+      - name: Install node dependencies
+        run: npm ci
+        shell: bash
+
+      - name: Run all tests
+        run: |
```

Comment:
I think there are advantages and disadvantages to this bash approach. It's a bit arbitrary though given that the list of tasks supplied in `npmTestScripts` needn't actually be test scripts (i.e. you could supply a value of `deploy` and end up running `npm run deploy` in a "test" job.

I think one of the disadvantages is that if the first job in sequence fails, you don't get test telemetry from subsequent jobs. In that regard, it might be better to run this as a matrix job.

The last piece of commentary would be that we typically haven't been this specific in past (where npm jobs needed to be specifically called). The `lint` script for example in some projects encapsulates multiple "sub-jobs":

```json
{
  "scripts": {
    "lint:ts": "eslint ...",
    "lint:docker": "hadolint ...",
    "lint": "npm run lint:ts && npm run lint:docker"
  }
}
```

This approach allows the `lint` script to be run by convention and the specifics of what gets linted is left to the project (via `package.json` configuration). This is more a vestige of our time using Makefiles, and might not be the best approach anymore – but if we do change course, we should probably work to have consistent interfaces across linters, tests, and other scripts that may have subcomponents.